portfolio/src/main/webapp/navbar.html

```diff
@@ -8,23 +8,29 @@
   <nav class="navbar navbar-expand-lg navbar-light">
     <div class="navbar-collapse bg-light" id="navbarColor03">
       <ul class="navbar-nav mr-auto">
-        <li class="nav-item">
+        <li id="index" class="nav-item">
           <a class="nav-link" href="index.html">HOME</a>
         </li>
-        <li class="nav-item">
+        <li id="about" class="nav-item">
           <a class="nav-link" href="about.html">ABOUT</a>
         </li>
-        <li class="nav-item">
+        <li id="projects" class="nav-item">
           <a class="nav-link" href="projects.html">PROJECTS</a>
         </li>
-        <li class="nav-item">
+        <li id="courses" class="nav-item">
           <a class="nav-link" href="courses.html">COURSES</a>
         </li>
-        <li class="nav-item">
+        <li id="comments" class="nav-item">
           <a class="nav-link" href="comments.html">COMMENTS</a>
         </li>
       </ul>
-      <div class="icon-bar">
+      <script>
+        let currPage = this.location.pathname == '/' ? '/index.html' : this.location.pathname;
+        currPage = currPage.slice(1, -5);
+        const currPageLink = document.getElementById(currPage);
+        currPageLink.className = 'nav-item active';
+      </script>
```

Comment:
looking for a way to not have to embed script in the html file

Reply:
Hm - maybe we can add an onclick to the list items. The onclick can take in the id and then in your js file you can write the functionality for the onclick to make all list items not active, except for the one clicked.

Reply:
Oh wait I see, you want it active if you're on that page. This might still work for that use case..

Reply:
I was having a hard time doing it that way because the nav bar reloads when you click on an item. I guess there would need to be some indicator that it had been clicked before the reload and as a result sets it to active after the reload

Reply:
Ah I see, yeah we need some way of saving the state of the nav bar. Maybe in the j query script when you load the navbar, we can grab the page link and add an active class state.

So something like this:
`<script> 
      $(function(){
        $("#navbar-content").load("navbar.html"); 
        $("grab the id of the page here").addClass('nav-item active'); 
      });
    </script>
`

Reply:
There's also this thing called event.preventDefault() which you may be able to use to stop the page reload from happening when clicking on other things. https://api.jquery.com/event.preventDefault/

Reply:
the on load one works but you have to have that in the load() callback : 
$("#navbar-content").load("navbar.html", function() {
    $("page id").addClass('nav-item active');
});

I'm still havine trouble getting it to work with the js click handler way because of the space in the class name....

Reply:
I think in this case it would be simplest to just put it in the callback. So load the navbar, and if we are on this current page, add the active class.

The reason we have to do this, is because there is no way to save 'state' (I think). Because we are fully reloading the page and the navbar each time. Modern frameworks can prevent this by comparing a virtual dom to the current dom and only renders the difference. Therefore your navbar wouldn't be reloaded each time and therefore could save and update its state. I'm a little shaky on this information so take it with a grain of salt... :)

Reply:
The callback function works! I just committed it here 29eabce , but if using a click handler is preferable let me know and I will work on that

Reply:
Thanks for bearing with us, good work on this!